sources/packages/backend/apps/api/src/route-controllers/overaward/_tests_/e2e/overaward.students.controller.getOverawardBalance.e2e-spec.ts

```diff
@@ -0,0 +1,66 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import { createFakeDisbursementOveraward } from "@sims/test-utils";
+import { Repository } from "typeorm";
+import {
+  DisbursementOveraward,
+  DisbursementOverawardOriginType,
+  Student,
+} from "@sims/sims-db";
+
+import {
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  FakeStudentUsersTypes,
+  getStudentByFakeStudentUserType,
+  getStudentToken,
+} from "../../../../testHelpers";
+
+describe("OverawardStudentsController(e2e)-getOverawardBalance", () => {
+  let app: INestApplication;
+  let student: Student;
+  let disbursementOverawardRepo: Repository<DisbursementOveraward>;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    student = await getStudentByFakeStudentUserType(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+      dataSource,
+    );
+    disbursementOverawardRepo = dataSource.getRepository(DisbursementOveraward);
+  });
+
+  it("Should return correct value for overaward balance", async () => {
+    // Create an overaward.
```

Comment:
In the PR there are 2 E2E files and they run in parallel and the data created by testFile1 may interfere the validation of testFile2 and vice versa because they all belong to the same student.
Even if we run them sequentially there will be a cascading effect of data created previously because of using single student.

I believe we must try to fake the implementation of  `const dbUser = await this.userService.getUserLoginInfo(userToken.userName);` in jwt.strategy.ts to proceed.